java/semantickernel-api/src/main/java/com/microsoft/semantickernel/memory/MemoryRecord.java

```diff
@@ -27,7 +27,7 @@ public class MemoryRecord extends DataEntryBase {
      */
     public MemoryRecord(
             @Nonnull MemoryRecordMetadata metadata,
-            @Nonnull Embedding<Float> embedding,
+            @Nonnull Embedding<? extends Number> embedding,
```

Comment:
The dotnet code says Embedding "Represents a strongly typed vector of numeric data", then goes on to say that TEmbedding is an unmanaged generic type. Shouldn't Embedding.cs declare `where TEmbedding : INumber<TEmbedding>`? Or maybe `IFloat<TEmbedding>` which still means that TEmbedding can be float or double.  If TEmbedding is always a float, then Embedding.java needs to change to `class Embedding<Float>`. At the moment, it is `class Embedding<EmbeddingType extends Number>` (EmbeddingType is strongly typed as a Number). 

I would happily change parameters that declare `Embedding<? extends Number>` to `Embedding<Float>`, but I think this should be done in a separate 'Fix Embedding generic type' PR since one should also fix `class Embedding<EmbeddingType extends Number>` as part of that effort.

Reply:
Hi @dsgrieve let me chase up with some folks on our side about why this is always a Float in the C# code. I'm OK with refactoring later as the change is going into the experimental branch.